src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -593,6 +540,11 @@ object OrderedRVD {
       .flatMap(_.samples)
       .sorted(pkOrd)
 
+    val min = pkis.map(_.min).min(pkOrd)
+    val max = pkis.map(_.max).max(pkOrd)
+
+    keys = min +: keys :+ max
```

Comment:
The min is the first and the max is the last because keys was just sorted.  I think min/max computation and adding the min/max to key is unnecessary.